crates/pink/sidevm/macro/src/macro_ocall.rs

```diff
@@ -0,0 +1,431 @@
+use heck::ToSnakeCase;
+use proc_macro2::{Ident, Literal, Span, TokenStream};
+use syn::{parse_quote, Result};
+
+struct OcallMethod {
+    id: i32,
+    fast_return: bool,
+    fast_input: bool,
+    args: Vec<TokenStream>,
+    method: syn::TraitItemMethod,
+}
+
+impl OcallMethod {
+    fn parse(method: &syn::TraitItemMethod) -> Result<Self> {
+        let mut id = None;
+        let mut fast_return = false;
+        let mut fast_input = false;
+
+        for attr in method.attrs.iter() {
+            if attr.is_ocall() {
```

Comment:
Don't bother to change, but in the next time, I'd adopt "early return" style code to make it fewer levels:

```rust
if !attr.is_ocall() {
  continue;
}
match ...
```

Reply:
Yeah, I really like the `early return` style too. Not sure how i wrote this.